### PR TITLE
Add tests for gradients when using HPUPrecisionPlugin

### PR DESCRIPTION
**Pull request by ankitgola005**
<details>
  <summary><b>Add tests for gradients when using HPUPrecisionPlugin</b></summary>

- [ ] Was this discussed/agreed via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

## What does this PR do?

Fixes # (issue).

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in GitHub issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/intermediate.rst [hunk 1]
```diff
@@ -106,6 +106,46 @@ These also allow for fine tuning with `enabled` for enabling and disabling mixed
...
+    def forward(self, x):
+        """Forward."""
+        # HPUPrecisionPlugin wraps a forward_context on train / val / predict / test _steps.
+        # This makes torch.autocast(enabled=True) as used in previous example redundant.
+        torch.hpu.is_autocast_hpu_enabled() # Returns True
+
+        with torch.autocast(device_type="hpu", dtype=torch.bfloat16, enabled=False):
+            torch.hpu.is_autocast_hpu_enabled() # Returns False
+            with torch.autocast(device_type="hpu", dtype=torch.bfloat16, enabled=False):
+                torch.hpu.is_autocast_hpu_enabled() # Returns True
```
Comment (jerome-habana) [inline docs/source/intermediate.rst:141]: why true here ?

Comment (ankitgola005) [reply docs/source/intermediate.rst:141]: Fixed

**Review by jerome-habana — APPROVED** [review]
